dom/node_element.go

```diff
@@ -45,15 +47,18 @@ func (n *Element) SetNamespace(uri, prefix string, activate ...bool) error {
 		return errors.New("missing uri for SetNamespace")
 	}
 
-	ns, err := clib.XMLNewNs(n, uri, prefix)
-	if err != nil {
-		return err
-	}
-
-	if activateflag {
-		if err := clib.XMLSetNs(n, wrapNamespaceNode(ns)); err != nil {
+	if oldNs, _ := clib.XMLSearchNs(doc, n, prefix); oldNs == 0 {
+		// Namespace not found, create a new one
+		ns, err := clib.XMLNewNs(n, uri, prefix)
+		if err != nil {
 			return err
 		}
+
+		if activateflag {
+			if err := clib.XMLSetNs(n, wrapNamespaceNode(ns)); err != nil {
+				return err
+			}
+		}
 	}
```

Comment:
I pulled your library and found the possible way to fix by using the same namespace pointer if it already exists.
I am not familiar with CGO, so I am not sure if this is the right way to do.
```suggestion
	ns, _ := clib.XMLSearchNs(doc, n, prefix)
	if ns == 0 {
		// Namespace not found, create a new one
		ns, err = clib.XMLNewNs(n, uri, prefix)
		if err != nil {
			return err
		}
	}

	if activateflag {
		if err := clib.XMLSetNs(n, wrapNamespaceNode(ns)); err != nil {
			return err
		}
	}
```